### PR TITLE
Move subtensor rewrites from basic opt to subtensor opt v2

### LINKED ISSUE #545: Move `*Subtensor*` rewrites from `basic_opt` into `subtensor_opt`
The `aesara.tensor.subtensor_opt` module was created recently and we should move all the `*Subtensor*` rewrites that are still in `aesara.tensor.basic_opt` into that module.

### PR DESCRIPTION
**Pull request by spiyer99**
**There is an informative high-level description of the changes.**
Moves the *Subtensor* rewrites from aesara.tensor.basic_opt to aesara.tensor.subtensor_opt.
Continuation of a previously submitted PR that became a bit too messy (#551).

**The description and/or commit message(s) references the relevant GitHub issue(s).**
Resolves #545.

**+ [ ] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).**
Yes

**The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).**
Yes

**The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.  If your commit description starts with "Fix...", then you're probably making this mistake.**
Yes

### CODE REVIEW

**Comment by ricardoV94** [comment]
That binary file seems to have cropped in again

File: aesara/tensor/basic_opt.py [hunk 1]
```diff
@@ -2472,384 +2303,6 @@ def local_upcast_elemwise_constant_inputs(fgraph, node):
 # Subtensor opts #
 ##################
```
Comment (ricardoV94) [inline aesara/tensor/basic_opt.py:2304]: These comment lines can be removed

**Review by brandonwillard — DISMISSED** [review]
These changes look good, thanks!

If you remove the added file, we can run the tests and finish the review.

File: aesara/tensor/basic_opt.py (hunk 1 above)
Comment (spiyer99) [reply aesara/tensor/basic_opt.py:2304]: @ricardov94, thanks for the suggestion. What’s the best way to change a previous commit in this case? Should I use `git reset HEAD~` or something like that?

**Comment by spiyer99** [comment]
> These changes look good, thanks!
> 
> If you remove the added file, we can run the tests and finish the review.


Thanks @brandonwillard! I’ve removed the binary file (sorry I need to be less careless with things like that).

File: aesara/tensor/basic_opt.py (hunk 1 above)
Comment (ricardoV94) [reply aesara/tensor/basic_opt.py:2304]: You can make a new commit and then squash it with the old one in interactive rebase

Comment (spiyer99) [reply aesara/tensor/basic_opt.py:2304]: Thanks will do. I'll also fix the error that the tests are reflecting.

**Comment by brandonwillard** [comment]
It looks like the last rebase might have removed the commits.

**Comment by spiyer99** [comment]
Yeah sorry I'll need to redo this. I don't know what I messed up.

**Comment by brandonwillard** [comment]
> Yeah sorry I'll need to redo this. I don't know what I messed up.

No problem; if you want to describe the `git` actions you're taking, we might be able to help.

**Comment by spiyer99** [comment]
Thanks! I couldn't get `git rebase -i` to work. So instead I used the logic [described here](https://stackoverflow.com/questions/927358/how-do-i-undo-the-most-recent-local-commits-in-git) to undo the previous commits and redo them with my changes added in. It probably wasn't the best thing to do in hindsight. 

Is there a way we could keep this PR open?

**Comment by spiyer99** [comment]
@ricardoV94, @brandonwillard I re-commited the work. I think I figured out what I did wrong earlier. 

@ricardoV94 I also removed the comment lines that you specified earlier.

**Comment by ricardoV94** [comment]
Looks good so far @spiyer99 

The last thing we need to do is move the related subtensor tests in `tests/tensor/test_basic_opt` to `tests/tensor/test_subtensor_opt`

**Comment by spiyer99** [comment]
@ricardoV94 - thanks for your help.  I moved the tests in the latest commit.

**Comment by ricardoV94** [comment]
> @ricardoV94 - thanks for your help.  I moved the tests in the latest commit.
 

Seems like you copied the tests to the new location but might not have removed them from the old one?

**Comment by spiyer99** [comment]
@ricardoV94 thanks - I've updated the commit now.

**Comment by spiyer99** [comment]
@brandonwillard I noticed you dismissed the review?

**Comment by brandonwillard** [comment]
> @brandonwillard I noticed you dismissed the review?

Yes, I dismissed my old review requesting changes because it was stale.

**Pull request merged**